### DISCUSSION TITLE
Only mask some parts of a value?

### DISCUSSION THREAD

**Discussion by bjansen** [Breus/json-masker#86] [Q&A] [answered]
Is it possible to customize how a value is masked, for example
```
some-customer-email@example.com -> so***************il@example.com
NL91 FAKE 0417 1643 00 -> NL91 ************** 00
```

**Comment by Breus** [comment]
@bjansen currently this is not possible, but are considering adding support for this. We are currently thinking of a way to implement this in the most generic way possible to support all sorts of masks (besides the multiple current masking options).

**Reply by Breus** [reply]
I've just opened [this DRAFT PR for an initial ADR about this idea](https://github.com/Breus/json-masker/pull/89/files)

**Reply by Breus** [reply, marked as answer]
[This PR](https://github.com/Breus/json-masker/pull/96) implements the requested feature @bjansen 

Updated (28.03.2024): available in 1.0.0-rc3
```
builder..maskStringsWith(ValueMaskers.withRawValueFunction(value -> ...))
```

**Reply by bjansen** [reply]
Wow thanks, looks great!

**Reply by gavlyukovskiy** [reply]
@bjansen the PR review is always welcome :) We want to release 1.0.0 in the near future, so we might be able to change some things before that.

This PR introduces email masker that can do exactly what you've described, but I'm a bit hesitant about IBAN masker as it requires a bit more thought around edge cases, so this one might come later if we or anyone else needs it.
With the new generic interface you can implement your own `ValueMasker` (or `ValueMaskers.withTextFunction`) to mask the IBAN (contributions are welcome btw).

**Reply by bjansen** [reply]
I'll give it a try maybe tomorrow, but I guess `ValueMaskers.withTextFunction` should be enough to implement custom masking. I don't need to mask IBANs in my particular case, I mentioned them in my original question because they appear in your README.

**Reply by Breus** [reply]
@bjansen FYI: version 1.0.0-rc3 was released today which includes your requested feature  :-)
